hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFileReader.java

```diff
@@ -210,15 +214,33 @@ private HoodieLogBlock readBlock() throws IOException {
   private HoodieLogBlock createCorruptBlock() throws IOException {
     log.info("Log " + logFile + " has a corrupted block at " + inputStream.getPos());
     long currentPos = inputStream.getPos();
-    long nextBlockOffset = scanForNextAvailableBlockOffset();
+    byte[] corruptedBytes;
+    int corruptedBlockSize = 0;
+    boolean done = true;
+    BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream);
+    do {
+      // read upto logblocksize to find next magic header
+      corruptedBytes = new byte[logBlockSize];
+      int numberOfBytesRead = bufferedInputStream.read(corruptedBytes, 0, logBlockSize);
+      int estimatedCorruptBlockSize = (int) scanForNextAvailableBlockOffset(corruptedBytes,
+          numberOfBytesRead);
+      if (numberOfBytesRead == logBlockSize && estimatedCorruptBlockSize == numberOfBytesRead) {
```

Comment:
This seems to be checking : we were able to read block size worth of byte and the byte after that matched up to a next available block offset?